package.json

```diff
@@ -11,23 +11,31 @@
     "build:css": "lessc ./src/css/custom-widget.less ./dist/custom-widget.css",
     "build:cssmin": "cleancss -o ./dist/custom-widget.min.css ./dist/custom-widget.css",
     "build": "npm run build:scripts && npm run build:css && npm run build:cssmin",
-    "prepare": "npm run build"
+    "prepare": "npm run build",
+    "test": "vitest",
+    "coverage": "vitest run --coverage",
+    "check:types": "tsc --noEmit"
   },
   "devDependencies": {
     "@types/shortid": "0.0.29",
-    "@typescript-eslint/eslint-plugin": "3.7.0",
-    "@typescript-eslint/parser": "3.7.0",
-    "clean-css-cli": "4.3.0",
-    "eslint": "7.5.0",
-    "eslint-config-prettier": "6.11.0",
-    "eslint-plugin-prettier": "3.1.4",
-    "less": "3.12.2",
-    "prettier": "2.0.5",
-    "shortid": "2.2.15",
-    "ts-loader": "8.0.1",
-    "typescript": "3.9.7",
-    "webpack": "4.43.0",
-    "webpack-cli": "3.3.12"
+    "@typescript-eslint/eslint-plugin": "5.38.0",
+    "@typescript-eslint/parser": "5.38.0",
+    "@vitest/coverage-c8": "0.23.4",
+    "c8": "7.12.0",
+    "clean-css-cli": "5.6.1",
+    "eslint": "8.23.1",
+    "eslint-config-prettier": "8.5.0",
+    "eslint-plugin-import": "2.26.0",
+    "eslint-plugin-prettier": "4.2.1",
+    "jsdom": "20.0.0",
+    "less": "4.1.3",
+    "prettier": "2.7.1",
+    "shortid": "2.2.16",
+    "ts-loader": "9.4.1",
+    "typescript": "4.8.3",
+    "vitest": "0.23.4",
+    "webpack": "5.74.0",
+    "webpack-cli": "4.10.0"
```

Comment:
Updated all libraries, added an extra plugin for ESLint, and added test + coverage libraries (`vitest` for tests, `c8` and `@vitest/coverage-c8` for coverage).